src/k4a_ros_device.cpp

```diff
@@ -742,6 +772,42 @@ k4a_result_t K4AROSDevice::getImuFrame(const k4a_imu_sample_t& sample, sensor_ms
     return K4A_RESULT_SUCCEEDED;
 }
 
+#if defined(K4A_BODY_TRACKING)
+k4a_result_t K4AROSDevice::getBodyMarker(const k4abt_body_t& body, MarkerPtr marker_msg, int jointType, ros::Time capture_time)
+{
+    k4a_float3_t position = body.skeleton.joints[jointType].position;
+	k4a_quaternion_t orientation = body.skeleton.joints[jointType].orientation;
+
+    marker_msg->header.frame_id = calibration_data_.tf_prefix_ + calibration_data_.depth_camera_frame_;
+    marker_msg->header.stamp = capture_time;
+
+    // Set the lifetime to 0.25 to prevent flickering for even 5fps configurations.
+    // New markers with the same ID will replace old markers as soon as they arrive.
+    marker_msg->lifetime = ros::Duration(0.25);
```

Comment:
Could the lifetime be set based on the camera framerate? At 30FPS, do you start to see marker "history" building up?

Reply:
There is no history building up because markers with the same id (combination of body + joint id) should overwrite old markers with that id.
I tried to use the fps before but it was flickering because sometimes rviz doesn't render at the exact same framerate, is slower, when also drawing pointcloud.

Reply:
I have screencaptured a video showcasing that no history is building up and no flickering happens.
https://drive.google.com/open?id=1Qqk-l9Nw6YdIQN5KTNG8a8jpGzF-s0mD

Reply:
Sounds good! Thanks for thinking this through.